### PR TITLE
Wemo Binding: Converted Groovy tests to Java

### LINKED ISSUE #4071: [Test Failures] Wemo Binding
Happend here: https://travis-ci.org/eclipse/smarthome/builds/265218318?utm_source=github_status&utm_medium=notification

```
assert that thing handles OnOff command correctly(org.eclipse.smarthome.binding.wemo.handler.test.WemoHandlerOSGiTest)  Time elapsed: 10.16 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is <0>
     but: was <1>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.eclipse.smarthome.binding.wemo.handler.test.WemoHandlerOSGiTest$_removeThing_closure11.doCall(WemoHandlerOSGiTest.groovy:172)
	at org.eclipse.smarthome.binding.wemo.handler.test.WemoHandlerOSGiTest$_removeThing_closure11.doCall(WemoHandlerOSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:220)
	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:191)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.wemo.handler.test.WemoHandlerOSGiTest.removeThing(WemoHandlerOSGiTest.groovy:171)
	at org.eclipse.smarthome.binding.wemo.handler.test.WemoHandlerOSGiTest.tearDown(WemoHandlerOSGiTest.groovy:63)
```

### LINKED ISSUE #4927: [Test failures] Wemo test failure
As seen here:
https://travis-ci.org/eclipse/smarthome/builds/329017654?utm_source=github_status&utm_medium=notification

```
assert that thing updates channels at start(org.eclipse.smarthome.binding.wemo.handler.test.WemoLightHandlerOSGiTest)  Time elapsed: 11.127 sec  <<< ERROR!
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1442)
	at java.util.HashMap$KeyIterator.next(HashMap.java:1466)
	at com.google.common.collect.AbstractMapBasedMultimap$WrappedCollection$WrappedIterator.next(AbstractMapBasedMultimap.java:486)
	at java.util.AbstractCollection.addAll(AbstractCollection.java:343)
	at java.util.HashSet.<init>(HashSet.java:120)
	at sun.reflect.GeneratedConstructorAccessor31.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:77)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:102)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:190)
	at org.eclipse.smarthome.binding.wemo.test.GenericWemoOSGiTest.getThingHandlers(GenericWemoOSGiTest.groovy:254)
	at sun.reflect.GeneratedMethodAccessor35.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.wemo.test.GenericWemoOSGiTest.getThingHandler(GenericWemoOSGiTest.groovy:241)
	at sun.reflect.GeneratedMethodAccessor29.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:368)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:66)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.wemo.test.GenericWemoLightOSGiTest$_removeThing_closure2.doCall(GenericWemoLightOSGiTest.groovy:121)
	at org.eclipse.smarthome.binding.wemo.test.GenericWemoLightOSGiTest$_removeThing_closure2.doCall(GenericWemoLightOSGiTest.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:54)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:243)
	at sun.reflect.GeneratedMethodAccessor14.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:223)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.binding.wemo.test.GenericWemoLightOSGiTest.removeThing(GenericWemoLightOSGiTest.groovy:120)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:137)
	at org.eclipse.smarthome.binding.wemo.handler.test.WemoLightHandlerOSGiTest.tearDown(WemoLightHandlerOSGiTest.groovy:60)
```

Full log: [log.txt](https://github.com/eclipse/smarthome/files/1631663/log.txt)

### PR DESCRIPTION
**Pull request by triller-telekom**
Within the conversion I also did the following:
- Made WemoHttpCall non-static to mock it inside tests
- All Handler inherit from AbstractWemoHandler which has a setter for WemoHttpCall
- Removed necessity for servlets in tests by checking the content sent to the server
- Removed dependecies to the ItemRegistry

Fixes #4927
Fixes #4071

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/.classpath [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
+	<classpathentry kind="src" path="src/main/java"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src/main/groovy"/>
 	<classpathentry exported="true" kind="con" path="GROOVY_DSL_SUPPORT"/>
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/.classpath]: is groovy support still required in any way?

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryOSGiTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import org.jupnp.model.ValidationException;
+import org.jupnp.model.meta.Device;
+
+/**
+ * Tests for {@link WemoDiscoveryService}.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ * @author Stefan Triller - Ported Tests from Groovy to Java
+ */
+
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryOSGiTest.java]: empty line

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryOSGiTest.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+public class WemoDiscoveryOSGiTest extends GenericWemoOSGiTest {
+
+    // UpnP service information
+    String SERVICE_ID = "basicevent";
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryOSGiTest.java]: can all the fields be `private`?

Comment (htreu) [reply extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryOSGiTest.java]: And constant definitions `private static final`

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryParticipantTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+ * Tests for {@link WemoDiscoveryParticipant}.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ * @author Stefan Triller - Ported Tests from Groovy to Java
+ */
+
+public class WemoDiscoveryParticipantTest {
+    UpnpDiscoveryParticipant participant = new WemoDiscoveryParticipant();
+
+    String DEVICE_UDN = GenericWemoOSGiTest.DEVICE_MANUFACTURER + "_3434xxx";
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/discovery/test/WemoDiscoveryParticipantTest.java]: also, fields private and constants `p s f`

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoHandlerOSGiTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+ * Tests for {@link WemoHandler}.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ * @author Stefan Triller - Ported Tests from Groovy to Java
+ */
+
+public class WemoHandlerOSGiTest extends GenericWemoOSGiTest {
+
+    // Thing information
+    String DEFAULT_TEST_CHANNEL = WemoBindingConstants.CHANNEL_STATE;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoHandlerOSGiTest.java]: again fields.

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+/**
+ * Tests for {@link WemoHandler}.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ * @author Stefan Triller - Ported Tests from Groovy to Java
+ */
+public class WemoHandlerTest {
+
+    ThingTypeUID THING_TYPE = WemoBindingConstants.THING_TYPE_INSIGHT;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoHandlerTest.java]: fields.

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoLightHandlerOSGiTest.java [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+/**
+ * Tests for {@link WemoLightHandler}.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ * @author Stefan Triller - Ported Tests from Groovy to Java
+ */
+public class WemoLightHandlerOSGiTest extends GenericWemoLightOSGiTest {
+
+    static final String GET_ACTION = "GetDeviceStatus";
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/handler/test/WemoLightHandlerOSGiTest.java]: private or protected?

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoLightOSGiTest.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    protected String DEFAULT_TEST_CHANNEL_TYPE = "Switch";
+
+    // UPnP service information
+    String BRIDGE_MODEL_NAME = WEMO_BRIDGE_ID;
+    protected String DEVICE_MODEL_NAME = WEMO_LIGHT_ID;
+    protected String SERVICE_ID = "bridge";
+    protected String SERVICE_NUMBER = "1";
+    protected String SERVLET_URL = DEVICE_CONTROL_PATH + SERVICE_ID + SERVICE_NUMBER;
+
+    Bridge bridge;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoLightOSGiTest.java]: private?

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoOSGiTest.java [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    String DEVICE_TYPE = "Test";
+    int DEVICE_VERSION = 1;
+    protected String DEVICE_FRIENDLY_NAME = "WeMo Test";
+    String DEVICE_URL = "http://127.0.0.1:" + ORG_OSGI_SERVICE_HTTP_PORT;
+    String DEVICE_DESCRIPTION_PATH = "/setup.xml";
+    protected String DEVICE_CONTROL_PATH = "/upnp/control/";
+
+    ChannelTypeUID DEFAULT_CHANNEL_TYPE_UID = new ChannelTypeUID(WemoBindingConstants.BINDING_ID + ":channelType");
+
+    ManagedThingProvider managedThingProvider;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoOSGiTest.java]: protected?

File: extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoOSGiTest.java [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+
+    // This port is included in the run configuration
+    int ORG_OSGI_SERVICE_HTTP_PORT = 9090;
+
+    // Thing information
+    protected String TEST_THING_ID = "TestThing";
+
+    // UPnP Device information
+    public static String DEVICE_UDN = "Test-1_0-22124";
+    String DEVICE_TYPE = "Test";
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.wemo.test/src/main/java/org/eclipse/smarthome/binding/wemo/internal/test/GenericWemoOSGiTest.java]: review all fields for access modifiers.

**Comment by triller-telekom** [comment]
Thanks for reviewing. Indeed these modifiers did not exist in groovy and the code compiled after adding the semicolon. :-) But you are right. I have added them and could even remove some other unnecessary left overs.

**Pull request merged**